knowledge-base/datagrid-maui-toggle-row-details-cell-selection.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Managing Row Details in DataGrid for MAUI on Cell or Row Selection
+description: Learn how to toggle row details in a DataGrid for MAUI application based on cell or row selection.
+type: how-to
+page_title: How to Display Row Details Based on Cell or Row Selection in DataGrid for MAUI
+slug: datagrid-maui-toggle-row-details-cell-selection
+tags: datagrid, maui, row details, cell selection, row selection, expandedrowdetails, selectionchanged
+res_type: kb
+---
+
+## Environment
+
+| Product | Version | Author |
+| --- | --- | --- |
+| DataGrid for .NET MAUI | 7.1.0 | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova) | 
+
+## Description
+
+In a DataGrid for MAUI, I want to show or hide row details based on cell selection without using the default `DataGridToggleRowDetailsColumn`. The goal is to use the `SelectionChanged` event to achieve this functionality. This KB article also answers the following questions:
+- How can I show row details on cell selection in a DataGrid?
+- What is the method to toggle row details visibility based on cell selection?
+- How to use SelectionChanged event for showing row details in DataGrid for MAUI?
+
+## Solution
+
+To display row details based on cell selection, you must first define a `RowDetailsTemplate`. For more information on how to set up a `RowDetailsTemplate`, refer to the [official documentation]({%slug datagrid-row-details-template%}).
```

Comment:
As the solution consists of a few steps, make sure to list these steps explicitly in a short format. Then, use the list to reference to the detailed explanation.
```suggestion
To display row details based on cell selection:

1. Define a `RowDetailsTemplate`. For more information on how to set up a `RowDetailsTemplate`, refer to the [Row Details Template documentation]({%slug datagrid-row-details-template%}).
1. Handle the selection:
   * To display the row details when a cell is selected, [handle the cell selection](#handling-cell-selection).
   * To display the row details when a row is selected, [handle the row selection](#handling-row-selection).
```